cabal-install/Distribution/Client/Dependency/Modular/Linking.hs

```diff
@@ -263,35 +280,45 @@ makeCanonical lg qpn@(Q pp _) i =
 -- because having the direct dependencies in a link group means that we must
 -- have already made or will make sooner or later a link choice for one of these
 -- as well, and cover their dependencies at that point.
-linkDeps :: [Var QPN] -> PP -> FlaggedDeps comp QPN -> UpdateState ()
-linkDeps parents pp' = mapM_ go
+linkDeps :: QPN -> [Var QPN] -> FlaggedDeps Component QPN -> UpdateState ()
+linkDeps parent = \parents deps -> do
```

Comment:
Why the RHS lambda?

Do we really need to call the two args `parent` and `parents`? Seems very easy to confuse.